store/tikv/region_cache.go

```diff
@@ -16,22 +16,35 @@ package tikv
 import (
 	"bytes"
 	"sync"
+	"time"
+	"sync/atomic"
 
-	log "github.com/Sirupsen/logrus"
 	"github.com/juju/errors"
+	log "github.com/Sirupsen/logrus"
 	"github.com/petar/GoLLRB/llrb"
 	"github.com/pingcap/kvproto/pkg/kvrpcpb"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/pd/pd-client"
 	goctx "golang.org/x/net/context"
 )
 
+const (
+	rcDefaultRegionCacheTTL	= time.Minute * 2
+)
+
+// CachedRegion encapsulates {Region, TTL}
+type CachedRegion struct {
+	region	*Region
```

Comment:
Why not rename `Region` to `CachedRegion` directly and add the new field there?

Reply:
Regions may be used to partition requests. For this kind of tasks, ttl is of no use. 
In fact, ttl is only used for cache.
So, I prefer defining a new structure CachedRegion here.

Reply:
The `Region` is a cached region with or without access time.
Use different type just to add the access time makes confusion.